stripe.go

```diff
@@ -462,32 +509,99 @@ func FormatURLPath(format string, params ...string) string {
 	return fmt.Sprintf(format, untypedParams...)
 }
 
-// GetBackend returns the currently used backend in the binding.
-func GetBackend(backend SupportedBackend) Backend {
-	switch backend {
+// GetBackend returns one of the library's supported backends based off of the
+// given argument.
+//
+// It returns an existing default backend if one's already been created.
+func GetBackend(backendType SupportedBackend) Backend {
+	var backend Backend
+
+	backends.mu.RLock()
+	switch backendType {
 	case APIBackend:
-		backends.mu.RLock()
-		ret := backends.API
-		backends.mu.RUnlock()
-		if ret != nil {
-			return ret
+		backend = backends.API
+	case UploadsBackend:
+		backend = backends.Uploads
+	}
+	backends.mu.RUnlock()
+	if backend != nil {
+		return backend
+	}
+
+	backend = GetBackendWithConfig(
+		backendType,
+		&BackendConfig{
+			HTTPClient:        httpClient,
```

Comment:
Good eye. Yeah, I went back in forth on this — I settled for what you see because I felt that the explicitness is somewhat advantageous compared to setting a bunch of `0`/`nil` and then having to follow the code down to `GetBackendWithConfig` to see what's going to end up in there.

I don't feel too strongly about it though, and am happy to go with the other way too. What do you think?

Reply:
I don't feel strongly either so I'm fine leaving as is